src/main/java/org/sagebionetworks/bridge/services/StudyService.java

```diff
@@ -350,4 +355,22 @@ private Study phaseTransition(String appId, String studyId, StudyPhase targetPha
         
         return study;
     }
+    
+    /**
+     * Retrieves the custom event IDs from a schedule. The returned event IDs will 
+     * have the "custom:" prefix removed. If scheduleGuid is null, will return
+     * an empty set.
+     */
+    protected Set<String> getCustomEventIdsFromSchedule(String appId, String scheduleGuid) {
+        Set<String> existingEventIds = new HashSet<>();
```

Comment:
very minor: could use ImmutableSet.of() here to avoid creating an object instance

Reply:
I have to leave it as mutable because the validator removes items from the set.